scalafix-core/shared/src/main/scala/scalafix/internal/rule/RemoveUnusedTerms.scala

```diff
@@ -0,0 +1,36 @@
+package scalafix.internal.rule
+
+import scala.meta._
+import scalafix.Patch
+import scalafix.SemanticdbIndex
+import scalafix.rule.RuleCtx
+import scalafix.rule.SemanticRule
+
+case class RemoveUnusedTerms(index: SemanticdbIndex)
+    extends SemanticRule(index, "RemoveUnusedTerms") {
+
+  private val unusedTerms = {
+    val UnusedLocalVal = """local (.*) is never used""".r
+    index.messages.toIterator.collect {
+      case Message(pos, _, UnusedLocalVal(_*))  =>
+        pos
+    }.toSet
+  }
+
+  private def isUnused(defn: Defn) =
+    unusedTerms.contains(defn.pos)
+
+  private def removeLiterals(rhs: Term): String =
+    rhs match {
+      case Lit(_) => ""
+      case r => r.syntax
```

Comment:
You're potentially modifying the syntax of the original rhs here, I would probably change approach and produce an Option[Patch]

Reply:
I propose that we only remove the necessary tokens (val/identifier/=) and leave the rhs unchanged, there may be other rules touching the tokens inside the rhs.